src/python/turicreate/data_structures/sframe.py

```diff
@@ -1603,13 +1603,18 @@ def read_json(cls,
         [3 rows x 1 columns]
         """
         if orient == "records":
-            g = SArray.read_json(url)
-            if len(g) == 0:
-                return SFrame()
-            if g.dtype != dict:
-                raise RuntimeError("Invalid input JSON format. Expected list of dictionaries")
-            g = SFrame({'X1':g})
-            return g.unpack('X1','')
+            if type(url)==str and url[0] !='{' :
+                g = SArray.read_json(url)
+                if len(g) == 0:
+                    return SFrame()
+                if g.dtype != dict:
+                    raise RuntimeError("Invalid input JSON format. Expected list of dictionaries")
+                g = SFrame({'X1':g})
+                return g.unpack('X1','')
+            elif type(url)==str and url[0] =='{' :
+                url=pandas.read_json(url)
+                url=SFrame(url)
+                return url
         elif orient == "lines":
```

Comment:
Can we support JSON strings with lines orient? If we can, I think that should be part of this pull request. If we can't support this we should mention that in the docstring.

Reply:
I think it should be possible. It looks like `pandas.read_json` has a similar `orient` parameter.